functions/lib/stripe/webhooks.py

```
@@ -148,6 +149,70 @@ def check_if_session_has_been_processed_already(
+    """
+    try:
+        orders_query = (
+            db.collection("Orders")
+            .where("stripePaymentIntentId", "==", payment_intent_id)
+            .limit(1)
+            .get()
+        )
+
+        if len(orders_query) == 0:
+            logger.warning(
+                f"No order found with payment intent ID: {payment_intent_id}"
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)